op-challenger/fault/position.go

```diff
@@ -30,6 +29,7 @@ func (p *Position) TraceIndex(maxDepth int) int {
 	return mid
 }
 
+// move goes to the left or right child in the generalized index tree.
```

Comment:
I'm not sure what you mean by "generalised index tree" here. `Position` isn't using generalised indices - it's using separate Depth and IndexAtDepth.  You can convert between the two easily but they are different concepts.

Reply:
I think they are different representations of the same concept. `generalized index = 1 << depth | IndexAtDepth`. Maybe generealized index tree isn't correct, but there is a tree that it's moving through.

Reply:
They are both ways of referring to nodes in a tree, but it's important not to confuse the two and a "generalised index tree" isn't really a thing.  You have a tree, and you can use generalised indices to refer to nodes.  I suspect the name we want is something like "bisection tree" but not really sure.  Would definitely be good to get some clear language agreed on this to make discussions easier.